lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/viewcontroller/RootPresenter.java

```diff
@@ -28,13 +29,18 @@ public RootPresenter() {
         this(new RootAnimator(), new LayoutDirectionApplier());
     }
 
+    public boolean setRootCalled() {
+        return setRootCalled;
+    }
+
     @VisibleForTesting
     public RootPresenter(RootAnimator animator, LayoutDirectionApplier layoutDirectionApplier) {
         this.animator = animator;
         this.layoutDirectionApplier = layoutDirectionApplier;
     }
 
     public void setRoot(ViewController appearingRoot, ViewController<?> disappearingRoot, Options defaultOptions, CommandListener listener, ReactInstanceManager reactInstanceManager) {
+        setRootCalled = true;
```

Comment:
It would be better to move this into `animateSetRootAndReportSuccess`, right before the `listener.onSuccess` calls. And you'll need 2 lines of it, one in each block.